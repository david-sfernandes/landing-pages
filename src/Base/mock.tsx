/* eslint-disable import/no-anonymous-default-export */
import links from '../components/NavLinks/mock';
import gridMock from "../components/GridText/mock";
import GridText from "../components/GridText";

export default {
  children: (
    <>
      <GridText {...gridMock}/>
      <GridText {...gridMock} background/>
    </>
  ),
  links: links,
  logoData: {
    text: 'Logo',
    link: '#'
  },
  footerHtml: '<p>Footer test</p>',
}