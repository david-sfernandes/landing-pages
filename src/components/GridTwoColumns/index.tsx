import Heading from "../Heading";
import SectionBackground from "../SectionBackground";
import TextComponent from "../TextComponent";
import Container, { Image, ImageContainer, TextContainer } from "./style";

export default function GridTwoColumns({
  title,
  text,
  srcImg,
  background = false,
}: GridTwoColumnsProps) {
  return (
    <SectionBackground background={background}>
      <Container>
        <TextContainer>
          <Heading uppercase darkColor={!background} as="h2">{title}</Heading>
          <TextComponent>{text}</TextComponent>
        </TextContainer>
        <ImageContainer>
          <Image src={srcImg} alt={title} />
        </ImageContainer>
      </Container>
    </SectionBackground>
  );
}
