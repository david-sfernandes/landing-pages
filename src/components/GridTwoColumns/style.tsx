import styled, { css } from "styled-components";
import * as Heading from "../Heading/style";

const Container = styled.div`
  ${({ theme }) => css`
    display: grid;
    grid-template-columns: 1fr 2fr;
    align-items: center;
    gap: ${theme.spacings.lg};

    @media ${theme.media.lteMedium} {
      grid-template-columns: 1fr;
      text-align: center;
    }

    ${Heading} {
      margin-bottom: ${theme.spacings.xl};
    }
  `}
`

export const TextContainer = styled.div`
   ${({ theme }) => css`
    @media ${theme.media.lteMedium} {
      margin-bottom: ${theme.spacings.lg};
    }
  `}
`

export const ImageContainer = styled.div``

export const Image = styled.img`
  width: 100%;
`

export default Container;