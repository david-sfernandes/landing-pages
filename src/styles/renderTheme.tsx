import { render } from "@testing-library/react";
import { ReactNode } from "react";
import { ThemeProvider } from 'styled-components';
import { theme } from "./theme";

export default function renderTheme(children: ReactNode) {
  return (
    render(<ThemeProvider theme={theme}>{children}</ThemeProvider>)
  )
}