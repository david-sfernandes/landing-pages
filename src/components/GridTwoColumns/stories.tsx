import GridTwoColumns from ".";
import mock from './mock';

const story = {
  title: "GridTwoColumns",
  component: GridTwoColumns,
  args: mock
};
export default story;

export const Template = (args: any) => <GridTwoColumns {...args} />;
