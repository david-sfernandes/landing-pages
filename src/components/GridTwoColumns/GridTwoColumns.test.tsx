import GridTwoColumns from '.';
import renderTheme from '../../styles/renderTheme';

import mock from './mock';

describe('<GridTwoColumns />', () => {
  it('should render two column grid', () => {
    const { container } = renderTheme(<GridTwoColumns {...mock} />);
    expect(container).toMatchSnapshot();
  });
});