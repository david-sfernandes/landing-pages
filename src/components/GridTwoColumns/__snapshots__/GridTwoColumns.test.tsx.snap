// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GridTwoColumns /> should render two column grid 1`] = `
.c5 {
  color: #0A1128;
  font-size: 56px;
  text-transform: uppercase;
}

.c1 {
  max-width: 1200px;
  margin: 0 auto;
  padding: 32px;
}

.c0 {
  background: #ffffff;
  color: #0A1128;
  min-height: 100vh;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c6 {
  font-size: 24px;
}

.c2 {
  display: grid;
  grid-template-columns: 1fr 2fr;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 32px;
  default: .c4;
}

.c2 {
  margin-bottom: 40px;
}

.c7 {
  width: 100%;
}

@media (max-width:768px) {
  .c5 {
    font-size: 32px;
  }
}

@media (max-width:768px) {
  .c2 {
    grid-template-columns: 1fr;
    text-align: center;
  }
}

@media (max-width:768px) {
  .c3 {
    margin-bottom: 32px;
  }
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <div
          class="c3"
        >
          <h2
            class="c4 c5"
          >
            Grid two columns
          </h2>
          <p
            class="c6"
          >
            Lorem ipsum dolor sit amet consectetur adipisicing elit. Distinctio magnam culpa eveniet doloribus harum? Ipsam, a necessitatibus? Sequi sunt accusantium quod, animi iure a, aliquid dolor ea vel magni dolore?
          </p>
        </div>
        <div
          class=""
        >
          <img
            alt="Grid two columns"
            class="c7"
            src="assets/images/javascript.svg"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
